api/src/domain/tax-filings/taxFilingsFactor.test.ts

```diff
@@ -0,0 +1,240 @@
+      get: jest.fn(),
+      put: jest.fn(),
+      findByEmail: jest.fn(),
+    };
+    taxFilingClient = taxFilingsFactory(stubUserDataClient, apiTaxFilingClient);
+  });
+
+  describe("lookup", () => {
+    describe("updates only state field when", () => {
+      const userData = generateUserData({
+        taxFilingData: generateTaxFilingData({
+          state: undefined,
+          filings: [generateTaxFiling({})],
+        }),
+      });
+
+      it("FAILED state", async () => {
+        apiTaxFilingClient.lookup.mockResolvedValue({
+          state: "FAILED",
+          filings: [],
+        } as TaxFilingLookupResponse);
+        stubUserDataClient.get.mockResolvedValue(userData);
+        expect(await taxFilingClient.lookup(userIdTaxIdBusinessName)).toEqual({
+          ...userData,
+          taxFilingData: { ...userData.taxFilingData, state: "FAILED" },
+        });
+      });
+
+      it("PENDING state", async () => {
+        apiTaxFilingClient.lookup.mockResolvedValue({
+          state: "PENDING",
+          filings: [],
+        } as TaxFilingLookupResponse);
+        stubUserDataClient.get.mockResolvedValue(userData);
+        expect(await taxFilingClient.lookup(userIdTaxIdBusinessName)).toEqual({
+          ...userData,
+          taxFilingData: { ...userData.taxFilingData, state: "PENDING" },
+        });
+      });
+
+      it("API_ERROR state", async () => {
+        apiTaxFilingClient.lookup.mockResolvedValue({
+          state: "API_ERROR",
+          filings: [],
+        } as TaxFilingLookupResponse);
+        stubUserDataClient.get.mockResolvedValue(userData);
+        expect(await taxFilingClient.lookup(userIdTaxIdBusinessName)).toEqual({
+          ...userData,
+          taxFilingData: { ...userData.taxFilingData, state: "API_ERROR" },
+        });
+      });
+    });
+
+    it("updates state and filing data on successful lookup", async () => {
+      const filingData = generateTaxFiling({});
+      const userData = generateUserData({
+        taxFilingData: generateTaxFilingData({
+          state: undefined,
+          filings: [generateTaxFiling({})],
+        }),
+      });
+      apiTaxFilingClient.lookup.mockResolvedValue({
+        state: "SUCCESS",
+        filings: [filingData],
+      } as TaxFilingLookupResponse);
+      stubUserDataClient.get.mockResolvedValue(userData);
+      expect(await taxFilingClient.lookup(userIdTaxIdBusinessName)).toEqual({
+        ...userData,
+        taxFilingData: { ...userData.taxFilingData, state: "SUCCESS", filings: [filingData] },
+      });
+    });
+  });
+
+  describe("onboarding", () => {
+    describe("returns lookup data", () => {
+      let userData: UserData;
+
+      beforeEach(() => {
+        userData = generateUserData({
+          taxFilingData: generateTaxFilingData({
+            state: undefined,
+            filings: [generateTaxFiling({})],
+          }),
+        });
+      });
+
+      it("when lookup response is API_ERROR", async () => {
```

Comment:
`it` tests should read as a sentence. Enforcing this is what allows us to enforce behavior-driven testing, because using the `it` sentence as a descriptor of what "it" (the thing under test) _does_ is making sure that you describe the _behavior_ of it.

So all these tests should describe not the SETUP of the test (ie, "it when the response is error") but instead the OUTCOME of the test (ie, "it does not call onboarding when...") or whatever the outcome may be